runtime/src/bank.rs

```diff
@@ -685,15 +700,6 @@ impl Bank {
 
             // freeze is a one-way trip, idempotent
             *hash = self.hash_internal_state();
-            true
-        } else {
-            false
-        }
-    }
-
-    pub fn freeze(&self) {
-        if self.set_hash() {
-            self.update_slot_hashes();
```

Comment:
Updating `slot_hashes` here results in mutating bank_hash of frozen `slot`, which complicates things. So just delay updating.